cEP-0031.md

```diff
@@ -0,0 +1,358 @@
+                message = '{} is conflicting package.'.format(conflict)
+                yield Result.from_values(origin=self, message=message,)
+```
+
+## Issues to be solved
+
+### Pycodestyle - PEP8Bear Integration
+
+Issues covered:
+
+- [coala/coala-bears#1897][1897]
+- [coala/projects#551][551]
+- [hhatto/autopep8#227][227]
+
+Since the autopep8 upstream is not likely to have any major improvements,
+we could have the improvements implemented to the PEP8Bear. Since autopep8 uses
+Pycodestyle for error detection, we could integrate the Pycodestyle with
+PEP8Bear.
+
+Doing this will involve invoking PEP8Bear after running PycodestyleBear.
+
+To implement this, we should be able to pass the selection of the issues
+through the `--line-range` and `--select` options available in autopep8.
+If multiple similar issues are available in the same line-range then we should
+implement the `--column-range` option in autopep8 which will allow us to
+pass the appropriate issue to generate a fix for.
+
+This will allow the user to generate a fix for only selected issues.
+
+```py
+import autopep8
+import logging
+
+from bears.python.PycodestyleBear import PycodestyleBear
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+from coalib.testing.LocalBearTestHelper import execute_bear
+from queue import Queue
+
+
+class PyPEP8Bear(LocalBear):
+    """
+    Integration of PycodestyleBear and PEP8Bear.
+    """
+    LANGUAGES = {'Python', 'Python 2', 'Python 3'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+
+    def run(self, filename, file,
+            max_line_length: int = 79,
+            indent_size: int = SpacingHelper.DEFAULT_TAB_WIDTH,):
+        """
+        Detects and fixes PEP8 incompliant code. This bear will not change
+        functionality of the code in any way.
+
+        :param max_line_length:
+            Maximum number of characters for a line.
+            When set to 0 allows infinite line length.
+        :param indent_size:
+            Number of spaces per indentation level.
+        """
+        if not max_line_length:
+            max_line_length = sys.maxsize
+
+        queue = Queue()
+        section = Section('')
+        section.append(Setting('max_line_length', max_line_length))
+
+        with execute_bear(PycodestyleBear(section, queue),
+                          filename, file,) as results:
+
+            for result in results:
+                start_line = result.affected_code[0].start.line
+                end_line = result.affected_code[0].end.line
+                # start_column = result.affected_code[0].start.column
+                # end_column = result.affected_code[0].end.column
```

Comment:
What happened with the column?

Reply:
Actually, they will be used for the `--column-range` which is not yet implemented in autopep8 (see https://github.com/hhatto/autopep8/pull/452). I tried testing the implementation there, but since it's not merged I commented this part out.